eng/build-native.sh

```diff
@@ -267,17 +267,12 @@ __ResultsDir=$__RootBinDir/TestResults/$__BuildType
 __PackagesBinDir=$__RootBinDir/packages/$__BuildType/Shipping
 __ExtraCmakeArgs="-DCLR_MANAGED_BINARY_DIR=$__RootBinDir/bin -DCLR_BUILD_TYPE=$__BuildType"
 __DotNetCli=$__ProjectRoot/.dotnet/dotnet
-__MSBuildPath=$__ProjectRoot/.dotnet/sdk/2.1.505/MSBuild.dll
 __DotNetRuntimeVersion=2.1.9
```

Comment:
Since you are upgrading the SDK to 3.0.x, this doesn't match what it meant to be.  The SOS tests assume that this variable is set to the SDK runtime version and assumes it is a 2.1.x runtime.